### PR TITLE
tailscale: use v2 client for device subnet routes

### PR DESCRIPTION
**Pull request by oxtoacart**
Updates tailscale/corp#21867

Acceptance test [passed](https://github.com/tailscale/corp/actions/runs/10308138358).

### CODE REVIEW

File: tailscale/resource_device_authorization_test.go [hunk 1]
```diff
@@ -12,18 +12,18 @@ import (
 	tsclient "github.com/tailscale/tailscale-client-go/v2"
 )
 
-func TestAccTailscaleDeviceAuthorization_Basic(t *testing.T) {
+func TestAccTailscaleDeviceAuthorization(t *testing.T) {
```
Comment (oxtoacart) [inline tailscale/resource_device_authorization_test.go:15]: Drive-by naming change

File: tailscale/resource_device_authorization_test.go [hunk 2]
```diff
@@ -12,18 +12,18 @@ import (
...
 	const testDeviceAuthorization = `
-	data "tailscale_device" "test_device" {
-		name = "%s"
-	}
-	
-	resource "tailscale_device_authorization" "test_authorization" {
-		device_id = data.tailscale_device.test_device.id
-		authorized = true
-	}`
+		data "tailscale_device" "test_device" {
```
Comment (oxtoacart) [inline tailscale/resource_device_authorization_test.go:19]: Drive-by whitespace formatting change

**Review by mpminardi — APPROVED** [review]
